### PR TITLE
Updating changelog for 1.4.0 release

### PR DESCRIPTION
**Pull request by ciaranmcnulty**
IMO this constitutes a minor release but YMMV

This is a subtle hint that phpspec would benefit from a newly tagged prophecy version ;-)

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+  * Fixed stringifying of hash containing one value (thanks @avant1)
+  * Improved clarity of method call expectation exception (thanks @dantleech)
+  * Add ability to specify which argument is returned in willReturnArgument (thanks @coderbyheart)
+  * Add more information to MethodNotFound exceptions (thanks @ciaranmcnulty)
+  * Support for mocking classes with methods that return references (thanks @edsonmedina)
+  * Improved object comparison (thanks @whatthejeff)
+  * Adopted '^' in composer dependencies (thanks @GrahamCampbell)
+  * Fixed non-typehinted arguments being treated as optional (thanks @whatthejeff)
+  * Magic methods are now filtered for keywords (thanks @seagoj)
+  * More readable errors for failure when expecting single calls (thanks dantleech)
```
Comment (stof) [inline CHANGES.md]: why no ``@`` here while you use it for others ?

**Comment by stof** [comment]
Please update the branch alias to 1.4.x-dev too btw

File: CHANGES.md (hunk 1 above)
Comment (ciaranmcnulty) [inline CHANGES.md]: Typo, fixed

**Comment by ciaranmcnulty** [comment]
@stof Updated the composer.json

**Pull request merged**